File: specs/my-w1/project-alpha/backend/app/schemas/ticket.py
"""Ticket Pydantic schemas."""
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from app.schemas.tag import TagResponse


class TicketBase(BaseModel):
    """Base ticket schema."""

    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = Field(None, max_length=10000)


class TicketCreate(TicketBase):
    """Schema for creating a ticket."""

    tag_ids: Optional[List[int]] = Field(default_factory=list)


class TicketUpdate(BaseModel):
    """Schema for updating a ticket."""

    title: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = Field(None, max_length=10000)


class TicketResponse(TicketBase):
    """Schema for ticket response."""

    id: int
    status: str
    tags: List["TagResponse"] = Field(default_factory=list)
    created_at: datetime
    updated_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class TicketListResponse(BaseModel):
    """Schema for paginated ticket list response."""

    tickets: List[TicketResponse]
    total: int
    limit: Optional[int] = None
    offset: int = 0

File: specs/my-w1/project-alpha/frontend/src/types/ticket.ts
import { Tag } from './tag'

export type TicketStatus = 'pending' | 'completed'

export interface Ticket {
  id: number
  title: string
  description?: string
  status: TicketStatus
  tags: Tag[]
  created_at: string
  updated_at?: string
  completed_at?: string
}

export interface TicketCreate {
  title: string
  description?: string
  tag_ids?: number[]
}

export interface TicketUpdate {
  title?: string
  description?: string
}

export interface TicketListResponse {
  tickets: Ticket[]
  total: number
  limit: number
  offset: number
}
